controllers/v1/reportsController.js

```diff
@@ -2043,6 +2043,229 @@ module.exports = class Reports {
     });
   }
 
+  async getParentInterviewResponses(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        if (!req.query.fromDate) {
+          throw "From Date is mandatory"
+        }
+        let fromDate = new Date(req.query.fromDate.split("-").reverse().join("-"))
+        let toDate = req.query.toDate ? new Date(req.query.toDate.split("-").reverse().join("-")) : new Date()
+        toDate.setHours(23, 59, 59)
+
+        if (fromDate > toDate) {
+          throw "From date cannot be greater than to date."
+        }
+
+        let fetchRequiredSubmissionDocumentIdQueryObj = {};
+        fetchRequiredSubmissionDocumentIdQueryObj["programExternalId"] = req.params._id
+        fetchRequiredSubmissionDocumentIdQueryObj["parentInterviewResponses"] = { $exists: true }
```

Comment:
change parentInterviewResponsesFieldArray -> parentInterviewResponsesArray

we will change these afterward.

Reply:
Resolved.